### PR TITLE
Add auth-backend-module-azure-easyauth-provider

### PR DESCRIPTION
**Pull request by yaegashi**
## Hey, I just made a Pull Request!

Add the azure-easyauth provider plugin for the new auth backend system.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by yaegashi** [comment]
@Rugvip @freben @camilaibs Thanks for reviewing this PR.

I have refactored the legacy auth-easyauth provider and I believe it's ready to merge.

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @yaegashi, haven't done a full review but just wanted to mention can you please make sure to update the related auth docs as well, these ones: https://github.com/backstage/backstage/blob/master/docs/auth/microsoft/azure-easyauth.md. All you should need to do is update the config example at the beginning with the proper `signIn.resolvers`

**Comment by yaegashi** [comment]
@awanlin Thanks, updated auth docs as requested.

**Comment by freben** [comment]
Took the liberty of rebasing this one after another pull request made it required to do so

File: plugins/auth-backend-module-azure-easyauth-provider/src/module.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  // As we rely on App Services to manage & validate the ID and Access Token headers
+  // Without that, this users can be trivially impersonated.
+  if (env.WEBSITE_SKU === undefined) {
+    throw new Error('Backstage is not running on Azure App Services');
+  }
+  if (env.WEBSITE_AUTH_ENABLED?.toLowerCase() !== 'true') {
+    throw new Error('Azure App Services does not have authentication enabled');
+  }
+  if (
+    env.WEBSITE_AUTH_DEFAULT_PROVIDER?.toLowerCase() !== 'azureactivedirectory'
```
Comment (freben) [inline plugins/auth-backend-module-azure-easyauth-provider/src/module.ts:65]: all of these should probably be `.toLocaleLowerCase('en-US')` but let's not block on that perhaps (it was just moved from elsewhere, can improve separately)

File: plugins/auth-backend-module-azure-easyauth-provider/src/module.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  ) {
+    throw new Error('Authentication provider is not Entra ID');
+  }
+  if (process.env.WEBSITE_AUTH_TOKEN_STORE?.toLowerCase() !== 'true') {
```
Comment (freben) [inline plugins/auth-backend-module-azure-easyauth-provider/src/module.ts:69]:
```suggestion
  if (env.WEBSITE_AUTH_TOKEN_STORE?.toLowerCase() !== 'true') {
```

again, not blocking

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍 

Let's :shipit:

File: plugins/auth-backend-module-azure-easyauth-provider/src/module.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  moduleId: 'azure-easyauth-provider',
+  register(reg) {
+    reg.registerInit({
+      deps: {
+        providers: authProvidersExtensionPoint,
+      },
+      async init({ providers }) {
+        validateAppServiceConfiguration(process.env);
+        providers.registerProvider({
+          providerId: 'azureEasyAuth',
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply plugins/auth-backend-module-azure-easyauth-provider/src/module.ts:38]: Added an extra note about this in the changeset though

**Comment by awanlin** [comment]
Thanks for including the doc updates @yaegashi 🚀

**Pull request merged**